data/transactions/logic/eval.go

```diff
@@ -3473,6 +3505,11 @@ func (cx *EvalContext) getLatestTimestamp() (uint64, error) {
 	return uint64(ts), nil
 }
 
+// GetApplicationAddress memoizes app.Address() across a tx group's evaluation
+func (ep *EvalParams) GetApplicationAddress(app basics.AppIndex) basics.Address {
+	return ep.getApplicationAddress(app)
+}
```

Comment:
I added a new public method and kept the original one to limit the diff of this PR. This method is used in `ledger/simulation/resources.go` to look up app addresses.

I'd be willing to remove the private method altogether if we think that's a good idea.

Reply:
here I lean towards removing the private method since that's what defacto happening.

Reply:
removed private method in a2e2c4b3987fb5ff730343e4f862bfc129db5b5a